From c2a1f88a2e80d0668a8047b149543f5641ee24df Mon Sep 17 00:00:00 2001
From: Xuefu Zhang <xzhang@Cloudera.com>
Date: Wed, 10 Feb 2016 13:15:57 -0800
Subject: [PATCH 0495/1164] CDH-36612: HIVE-12951: Reduce Spark executor
 prewarm timeout to 5s (reviewed by Rui)

Conflicts:
	ql/src/java/org/apache/hadoop/hive/ql/exec/spark/RemoteHiveSparkClient.java

Change-Id: Ie36e3c818dfa6a27e0a4c5e5bbb8496cc5530973
---
 .../hive/ql/exec/spark/RemoteHiveSparkClient.java  |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/RemoteHiveSparkClient.java b/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/RemoteHiveSparkClient.java
index c10205d..43bc766 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/RemoteHiveSparkClient.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/exec/spark/RemoteHiveSparkClient.java
@@ -72,7 +72,7 @@
 
   private static final String MR_JAR_PROPERTY = "tmpjars";
   private static final transient Log LOG = LogFactory.getLog(RemoteHiveSparkClient.class);
-  private static final long MAX_PREWARM_TIME = 30000; // 30s
+  private static final long MAX_PREWARM_TIME = 5000; // 5s
   private static final transient Splitter CSV_SPLITTER = Splitter.on(",").omitEmptyStrings();
 
   private transient Map<String, String> conf;
@@ -115,10 +115,11 @@ private void createRemoteClient() throws Exception {
           LOG.info("Finished prewarming Spark executors. The current number of executors is " + curExecutors);
           return;
         }
-        Thread.sleep(1000); // sleep 1 second
+        Thread.sleep(500); // sleep half a second
       } while (System.currentTimeMillis() - ts < MAX_PREWARM_TIME);
 
-      LOG.info("Timeout (60s) occurred while prewarming executors. The current number of executors is " + curExecutors);
+      LOG.info("Timeout (" + MAX_PREWARM_TIME + 
+          "s) occurred while prewarming executors. The current number of executors is " + curExecutors);
     }
   }
 
-- 
1.7.9.5

